src/firefly/java/edu/caltech/ipac/firefly/server/packagedata/ZipHandler.java

```diff
@@ -177,32 +179,12 @@ private static InputStream getInputStream(String filename, FileInfo fi, File bas
                 URLConnection uc = URLDownload.makeConnection(url, cookies, headers);
                 uc.setRequestProperty("Accept", "text/plain");
 
-                String suggestedFilename = URLDownload.getSugestedFileName(uc);
-
-                //fallback to filename from URL path if content-disposition header is missing
-                if (isEmpty(suggestedFilename)) {
-                    String urlPath = url.getPath();
-                    suggestedFilename = urlPath.substring(urlPath.lastIndexOf('/') + 1);
-                }
-
-                //strip surrounding quotes, if any
-                suggestedFilename = suggestedFilename.replaceAll("^\"|\"$", "");
-
-                if (fi.hasFileNameResolver()) {
-                    fi.setExternalName(fi.resolveFileName(suggestedFilename));
-                } else {
-                    String extName = fi.getExternalName();
-
-                    if (!isEmpty(extName)) {
-                        if (extName.endsWith("/")) {
-                            //treat as directory, append suggested filename
-                            fi.setExternalName(extName + suggestedFilename);
-                        }
-                        //else: do nothing, extName is already a valid file path
-                    } else {
-                        //if no externalName set, use the suggested file name, from content-disposition or the URL path
-                        fi.setExternalName(suggestedFilename);
-                    }
+                String extName = ofNullable(fi.getExternalName()).orElse("").trim();
+                if (extName.isEmpty() || extName.endsWith("/")) {
+                    String suggestedFilename = URLDownload.getSugestedFileName(uc);
+                    suggestedFilename = isEmpty(suggestedFilename) ? getFileNameFromUrl(url) : suggestedFilename;
+                    suggestedFilename = suggestedFilename.replaceAll("[^a-zA-Z0-9._-]", "_");
```

Comment:
Move this line into `getFileNameFromUrl()`.

Reply:
done, addressed the other changes as well. Rebuilding the test builds to make sure no new bugs were introduced.